Feature/changelog (#465)
rsksmart/devportal: kb/faqs.md

### What is a smart contract?

> Smart contracts are contracts whose terms are encoded in computer language instead of legal language. Smart contracts can be executed by a computing network such as RSK, so that the terms of the contracts are automatically enforced by a protocol that all [nodes](https://en.wikipedia.org/wiki/Node_(networking)) in the network follow.

Reviewer: ```suggestion
Smart contracts are contracts whose terms are encoded in computer language instead of legal language. Smart contracts can be executed by a computing network such as RSK, so that the terms of the contracts are automatically enforced by a protocol that all nodes in the network follow.
```

### Is RSK centralised, federated, or decentralised?

> The RSK blockchain is highly  decentralised. RSK is merge-mined with Bitcoin, and has a hashpower that is second only to Bitcoin. As such, we believe it to be the most secure and censorship resistant smart contract platform; and the second most secure blockchain platform. Refer to [RSK Stats](https://stats.rsk.co/) for the live value of the RSK hash rate.

Reviewer: ```suggestion
The RSK blockchain is highly decentralised. RSK is merge-mined with Bitcoin, and has a hashpower that is second only to Bitcoin. As such, we believe it to be the most secure and censorship resistant smart contract platform; and the second most secure blockchain platform. Refer to [RSK Stats](https://stats.rsk.co/) for the live value of the RSK hash rate.
```

*How many nodes does a healthy protocol need?*

> Decentralization should not be measured only by the number of nodes but also about the diversity and independence of nodes. A few hundred independent RSK nodes is enough to serve a global cryptocurrency network at this stage, but we must not feel confident by that metric. Our objective is that full nodes are run by a diverse set of individuals, organizations and companies. That is the true meaning of decentralization: **don’t trust, verify** yourself. IOVLabs innovation and research area has developed several decentralized incentivizations mechanisms that may one day be integrated into full nodes.  Also, we’ve put great effort to reduce the resource consumption of full nodes, such as the [Unitrie proposal]( https://www.rsk.co/noticia/towards-higher-onchain-scalability-with-the-unitrie/) so that individuals can run nodes in standard laptops. Finally, we proposed a new technique for light clients [RSK Improvement Proposal](https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP45.md), to onboard those users running nodes in mobile phones. In summary, we’re making sure that the network remains healthy and decentralized in the future, both in node quantity and quality.

Reviewer: ```suggestion
Decentralization should not be measured only by the number of nodes but also about the diversity and independence of nodes. A few hundred independent RSK nodes is enough to serve a global cryptocurrency network at this stage, but we must not feel confident by that metric alone. Our objective is that full nodes are run by a diverse set of individuals, organizations and companies. That is the true meaning of decentralization: **Don’t trust, verify** yourself. IOVLabs' innovation and research area has developed several decentralized incentivizations mechanisms that may one day be integrated into full nodes.  Also, we’ve put great effort to reduce the resource consumption of full nodes, such as the [Unitrie proposal]( https://www.rsk.co/noticia/towards-higher-onchain-scalability-with-the-unitrie/) so that individuals can run nodes in standard laptops. Finally, we proposed a new technique for light clients [RSK Improvement Proposal](https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP45.md), to onboard those users running nodes in mobile phones. In summary, we’re making sure that the network remains healthy and decentralized in the future, both in node quantity and quality.
```

### Is RSK secure from selfish mining?

> RSK uses the DECOR+ consensus protocol. DECOR+  is incentive-compatible and protects the network from selfish-mining  when the rate of honest uncle blocks produced by the network is low. If the uncle rate is high, then a selfish incentive may arise, as described by Camacho-Lerner. To improve it, several fixes have been proposed, such as, the “sticky” rule, delaying the transfer of the weight of uncles in GHOST, or allowing referencing uncle-children in the same way as uncles. With any of these fixes, RSK consensus protocol becomes incentive-compatible assuming that transaction fees are stable, and there are no off-chain payments or bribes to miners.

Reviewer: ```suggestion
RSK uses the DECOR+ consensus protocol. DECOR+  is incentive-compatible and protects the network from selfish-mining  when the rate of honest uncle blocks produced by the network is low. If the uncle rate is high, then a selfish incentive may arise, as described by [Camacho-Lerner](https://scalingbitcoin.org/papers/DECOR-LAMI.pdf). To improve it, several fixes have been proposed, such as, the “sticky” rule, delaying the transfer of the weight of uncles in GHOST, or allowing referencing uncle-children in the same way as uncles. With any of these fixes, RSK consensus protocol becomes incentive-compatible assuming that transaction fees are stable, and there are no off-chain payments or bribes to miners.
```

This suite of protocols aim to solve the major problems that stop decentralized blockchain networks (ie: Bitcoin, RSK, Ethereum, etc.) from reaching mass adoption. From our point of view, the two main impediments are sustainable scaling (onchain scaling is possible but leads to higher maintenance cost for validation nodes and therefore to centralization) and developer usability (it can take several months for developers to learn how to use the technology and even mastering the tech, it’s very inefficient to build decentralized apps for the lack of a higher level protocol and reusable components).

> Following the guidelines of RIF OS, a series of blockchain based P2P platforms are being built using RNS; an implementation of RIF Directory on RSK, the first to be launched. RIF Lumino, the first implementation of RIF Payments is launched, for more information and how to set up a Lumino node, visit [RIF Lumino](https://developers.rsk.co/rif/lumino/). We want to emphasise more on the utility of the RIF token within the RIF OS ecosystem.

Reviewer: ```suggestion
Following the guidelines of RIF OS, a series of blockchain based P2P platforms are being built using RNS; an implementation of RIF Directory on RSK, the first to be launched. RIF Lumino, the first implementation of RIF Payments is launched, for more information and how to set up a Lumino node, visit [RIF Lumino](https://developers.rsk.co/rif/lumino/).
We wish to emphasise the utility of the RIF token within the RIF OS ecosystem.
```

### How can I recover my token from my jaxx wallet which was the only one that had RSK addresses, when the RSK faucet existed?

> Although it doesn’t happen very often, we regularly do planned fresh restarts of the RSK Testnet blockchain. This means that all account balances go to zero. A Testnet reset has been recently executed, so there is no way to recover Testnet funds once this is done. RSK Faucet still exists and you can get [Testnet RBTCs]( https://faucet.testnet.rsk.co/) and also [Testnet RIF - tRIF](https://faucet.rifos.org/).

Reviewer: ```suggestion
Although it doesn’t happen very often, we do planned fresh restarts of the RSK Testnet blockchain. This means that all account balances go to zero. A Testnet reset has been recently executed, so there is no way to recover Testnet funds once this is done. RSK Faucet still exists and you can get [Testnet RBTCs]( https://faucet.testnet.rsk.co/) and also [Testnet RIF - tRIF](https://faucet.rifos.org/).
```

We are working on new RIF Payments components to be launched soon as well as RIF Storage Protocol. By the end of the year we plan to count with a full suite of RIF OS services that will showcase how the full stack will work together.

## PowPeg Federation

### What is the PowPeg Federation?

> The RSK platform was launched with a federation of well-known and respected community members (blockchain companies with high security standards) (the Federation). Each federation member also referred to as a functionary or notary, is identified by a set of public keys, one used for a Bitcoin multi-signature, and others used for authentication and private communications. The conditions to become a Federation member have been established, including security policies, backup procedures, and legal requirements.  In 2020, the Federation migrated to a new security model called PowPeg. Under this system, a functionary does not have access to their multisig private key, which is controlled by an HSM, while they do maintain control of other authentication private keys.

Reviewer: ```suggestion
The RSK platform was launched with a federation of well-known and respected community members (blockchain companies with high security standards) (the Federation). Each federation member, also referred to as a functionary or notary, is identified by a set of public keys, one used for Bitcoin multi-signatures, and others used for authentication and private communications. The conditions to become a Federation member have been established, including security policies, backup procedures, and legal requirements.  In 2020, the Federation migrated to a new security model called PowPeg. Under this system, a functionary does not have access to their multisig private key, which is controlled by an HSM, while they do maintain control of other authentication private keys.
```

### How secure is RSK merge-mining?

Nakamoto consensus can be proven secure as long as there is an honest majority of miners. In case of a dishonest majority, malicious miners may find cheating a rational or irrational strategy depending on assumptions on how the market and the community would react to a 51% attack. RSK is no different from Bitcoin in theory, but in practice not all Bitcoin miners participate in merge-mining, so the requirement of honest hashrate would be higher. A ad-hoc monitoring system, called Armadillo, lowers the requirement by warning nodes if more than 50% of the miners turn malicious before malicious miners can do harm.

> - In terms of cryptographic security, RSK merge-mining uses both SHA256 and Keccak. Since there are no known practical attacks on these hashing functions, RSK is currently as secure as Bitcoin mining. In particular RSK assumes a stronger property from SHA256, which is that, it must not allow a “freestart collision” at a cost lower than 2^80 operations. This is because the RSK network uses a property of the Merkle–Damgård construction to compress the size of the SPV proof.

Reviewer: ```suggestion
In terms of cryptographic security, RSK merge-mining uses both SHA256 and Keccak. Since there are no known practical attacks on these hashing functions, RSK is currently as secure as Bitcoin mining. In particular RSK assumes a stronger property from SHA256, which is that, it must not allow a “freestart collision” at a cost lower than 2^80 operations. This is because the RSK network uses a property of the Merkle–Damgård construction to compress the size of the SPV proof.
```

There are only two other Bitcoin sidechain projects that are currently active: Liquid and Truthcoin’s drivechain. Liquid is a federated sidechain, somehow similar to RSK. Liquid aims to be an inter-exchange settlement network linking together cryptocurrency exchanges, enabling faster Bitcoin transactions. It’s optimized for a single use case. RSK is much more generic and programmable, having stateful smart-contracts. Also RSK is highly compatible with Ethereum applications, libraries and toolchains. It has a large ecosystem and trained developers. Liquid applications currently depend on a single library provided by Blockstream, and has a niche ecosystem.

> Another key difference is that Liquid uses its PowPeg Federation for block consensus, while RSK uses merge-mining, and currently it has about 40% of Bitcoin’s hashrate. Therefore RSK has actual “thermodynamic” security. Anyone can participate in RSK merge-mining, so anyone can receive transaction fees.

Reviewer: ```suggestion
Another key difference is that Liquid uses its Federation for block consensus, while RSK uses merge-mining, and currently it has about 40% of Bitcoin’s hashrate. Therefore RSK has actual “thermodynamic” security. Anyone can participate in RSK merge-mining, so anyone can receive transaction fees.
```

> Regarding onchain transaction throughput, RSK can achieve a higher volume than Liquid because essentially RSK’s payment transactions are smaller than Liquid’s. However, currently the transaction throughput in RSK is limited by its miners, which can increase or decrease the block gas limit. In following RSK network upgrades we may see the two important developments implemented: the LTCP protocol (see [RSKIP53](https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP53.md)) and parallel transaction processing (see [RSKIP04](https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP04.md)). These improvements together enable 30x transaction throughput increase in RSK. Another key difference between RSK and Liquid is that the RSK peg is open. It can be used by individual users without going through an exchange, and a KYC process. However, the fastest way to get RBTC is still exchanging BTC at a crypto exchange, because it takes a day to transfer bitcoins to RSK using the peg. In terms of PowPeg Federation security, Liquid uses a 11-out-of-15 multisig with a 2-of-3 time-locked emergency spend, and RSK uses a 8-out-of-15 multisig, so each sidechain has different traseofs between availability and security.

Reviewer: ```suggestion
Regarding onchain transaction throughput, RSK can achieve a higher volume than Liquid because essentially RSK’s payment transactions are smaller than Liquid’s. However, currently the transaction throughput in RSK is limited by its miners, which can increase or decrease the block gas limit. In following RSK network upgrades we may see the two important developments implemented: The LTCP protocol (see [RSKIP53](https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP53.md)) and parallel transaction processing (see [RSKIP04](https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP04.md)). These improvements together enable 30x transaction throughput increase in RSK. Another key difference between RSK and Liquid is that the RSK peg is open. It can be used by individual users without going through an exchange, and a KYC process. However, the fastest way to get RBTC is still exchanging BTC at a crypto exchange, because it takes a day to transfer bitcoins to RSK using the peg. In terms of PowPeg Federation security, Liquid uses a 11-out-of-15 multisig with a 2-of-3 time-locked emergency spend, and RSK uses a 8-out-of-15 multisig, so each sidechain has different trade offs between availability and security.
```
